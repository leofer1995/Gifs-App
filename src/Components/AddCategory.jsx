import React, { useState } from "react";
import PropTypes from 'prop-types';

const AddCategory = ({setCategories})=>{
    const [input, setInput] = useState('');

    const handleChange = (e)=>{
        setInput(e.target.value)
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        if(input.trim().length > 2){
            setCategories(catg => [input,...catg])
            setInput('')
        }
        
    }

    return (
        <form onSubmit={e=>{handleSubmit(e)}}>
            <input 
                className='searchBar'
                type="text" 
                value = {input}
                onChange = {e=>handleChange(e)}
            />
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;