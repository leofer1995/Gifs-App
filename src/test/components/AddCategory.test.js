import { shallow } from "enzyme";
import AddCategory from "../../Components/AddCategory";


describe('pruebas en el componente AddCategory',()=>{
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{//ciclos de vida de los test
        jest.clearAllMocks();//limpiar las simulaciones
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);//REINICIO

    })
    test('deberia mostrarse corewectamente',()=>{
        expect(wrapper).toMatchSnapshot()
    });

    test('debe de cambiar la caja de texto',()=>{
        //busco el input 
        const input = wrapper.find('input');
        //valor del value del elemento que dispara el evento 
        const value = 'Hola Mundo'
        //simular un evento change y ese evento va a tener la propiedad target.value que sera igual al value anterior
        input.simulate('change', {target:{value}});
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(value)

    });

    test('no deberia postear la infurmacion con submit',()=>{
        const form = wrapper.find('form');
        form.simulate('submit',{preventDefault:()=>{}});
        //No debio llamar la funcion setCategories
        expect(setCategories).not.toHaveBeenCalled();//BEFOREEACH
    });

    test('deberia llamar el setCategories y limpiar la caja de texto',()=>{
        const input = wrapper.find('input');
        input.simulate('change',{target:{value:'Lo Logre'}});
        const form = wrapper.find('form');
        form.simulate('submit',{preventDefault:()=>{}});
        expect(setCategories).toHaveBeenCalled();//se llamo 
        expect(setCategories).toHaveBeenCalledTimes(1);// se debe llamar una vez
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));//debe llamarse con una funcion
        //console.log(input.props())
        expect(input.props().value).toBe('')

    })
});