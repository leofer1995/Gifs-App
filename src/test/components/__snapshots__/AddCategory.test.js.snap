// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente AddCategory deberia mostrarse corewectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    className="searchBar"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
